include/engine/phantom_node.hpp

```diff
@@ -167,8 +167,8 @@ struct PhantomNode
     util::Coordinate location;
     util::Coordinate input_location;
     unsigned short fwd_segment_position;
+
     // is phantom node valid to be used as source or target
-  private:
```

Comment:
To have access to is_valid_* flags. Current ctor is not convenient for us to set this flags. If it's important, we can leave them private, but add some modify functions.